pkg/metadata/metadata.go

```diff
@@ -99,6 +92,10 @@ func init() {
 	mdctx = mdctx.WithProtocol(EqualsClaimID, func() ipnimd.Protocol { return &EqualsClaimMetadata{} })
 	mdctx = mdctx.WithProtocol(LocationCommitmentID, func() ipnimd.Protocol { return &LocationCommitmentMetadata{} })
 	MetadataContext = mdctx
+	// WithProtocol creates a _new_ context. Assign it to Default so claim
+	// metadata can be decoded elsewhere by refrencing libipni metadata.Default
+	// i.e. ipni-publisher
+	ipnimd.Default = mdctx
```

Comment:
Bug - see comment. The `ipni-publisher` store uses `metadata.Default` and was not able to decode our custom metadatas. I think ideally we'd pass metadata context through?

Reply:
yea we should